photon-ml/src/test/scala/com/linkedin/photon/ml/algorithm/CoordinateDescentTest.scala

```diff
@@ -76,19 +99,33 @@ class CoordinateDescentTest {
       doReturn(score).when(coordinate).score(model)
       doReturn((model, tracker)).when(coordinate).updateModel(model)
       doReturn((model, tracker)).when(coordinate).updateModel(model, score)
+
+      // Validation evaluator mock setup
+      doReturn(validationScore).when(validationScore). +(validationScore)
+      doReturn(modelScores).when(validationScore).scores
+      doReturn(validationScore).when(model).score(validationData)
+      doReturn(validationScore).when(validationScore).setName(Matchers.anyString())
+      doReturn(validationScore).when(validationScore).persistRDD(StorageLevel.INFREQUENT_REUSE_RDD_STORAGE_LEVEL)
     }
 
     // Run coordinate descent
-    val coordinateDescent = new CoordinateDescent(coordinates, evaluator, None, logger)
-    coordinateDescent.run(numIterations, gameModel)
+    val validationDataAndEvaluators = if (validationEvaluators.isEmpty) None else Option(validationData, validationEvaluators)
+    val coordinateDescent = new CoordinateDescent(coordinates, lossEvaluator, validationDataAndEvaluators, logger)
+    coordinateDescent.run(iterationCount, gameModel)
 
     // Verify the calls to updateModel
     if (coordinates.length == 1) {
-      verify(coordinates.head._2, times(numIterations)).updateModel(models.head)
-
+      verify(coordinates.head._2, times(iterationCount)).updateModel(models.head)
     } else {
       (coordinates zip models).map { case ((coordinateId, coordinate), model) =>
-        verify(coordinate, times(numIterations)).updateModel(model, score)
+        verify(coordinate, times(iterationCount)).updateModel(model, score)
+      }
+    }
+
+    // Verify the calls to the validation evaluator(s), if any
+    validationDataAndEvaluators.map { case (data, evaluators) =>
+      evaluators.map { case (evaluator) =>
+        verify(evaluator, times(iterationCount * coordinateCount)).evaluate(modelScores)
```

Comment:
We need a way to identify that the result is correct though - i.e. make sure that the returned **mock** model corresponds to the best evaluation.